liquidctl/driver/aquacomputer.py

```diff
@@ -354,13 +362,25 @@ def set_speed_profile(self, channel, profile, **kwargs):
         elif self._device_info["type"] == self._DEVICE_FARBWERK360:
             raise NotSupportedByDevice()
 
+    def _fan_name_to_hwmon_names(self, channel):
+        if "hwmon_ctrl_mapping" in self._device_info:
+            # Custom fan name to hwmon pwmX translation
+            pwm_name = self._device_info["hwmon_ctrl_mapping"][channel]
+        else:
+            # Otherwise, assume that fanX translates to pwmX
+            pwm_name = f"pwm{channel[3]}"
+
+        return pwm_name, f"{pwm_name}_enable"
+
     def _set_fixed_speed_hwmon(self, channel, duty):
-        hwmon_pwm_name = self._device_info["hwmon_ctrl_mapping"][channel]
-        hwmon_pwm_enable_name = f"{hwmon_pwm_name}_enable"
+        hwmon_pwm_name, hwmon_pwm_enable_name = self._fan_name_to_hwmon_names(channel)
 
         # Set channel to direct percent mode
         self._hwmon.write_int(hwmon_pwm_enable_name, 1)
 
+        # Some devices (Octo, Quadro and Aquaero) can not accept reports in quick succession, so slow down a bit
+        time.sleep(0.5)
```

Comment:
Is a delay always necessary?  Sleeping 500 ms would be quite noticiable: could we optimistically try without it, and only do it when necessary?

(Does it matter whether the previous write was redundant (i.e. when  `pwmX_enable` was already set)?)

Reply:
I've set the delay to 0.2s now. I've been spamming it and it still behaved correctly, though 0.5s wasn't _that_ noticeable, at least to me. Setting fan speed now takes about 0.3s as a whole, which feels instantaneous (it was ~0.1s previously).

There's no way to do it optimistically, it may accept setting the speed twice and fail the third time, out of the blue. And we can't read the `pwmX_enable` value either, because that also triggers sending HID reports to the device to read the state (aside from sensor reports, the driver does not cache anything).

Reply:
What I previously had in mind was something like:

```py
# pseudo-code, I don't know the exact failure mode when the device isn't ready
try:
    self.device.send_feature_report(ctrl_settings)
except IOError:
    # Maybe the device couldn't handle the quick succession of reports; try one more time after a bit.
    time.sleep(0.5)
    self.device.send_feature_report(ctrl_settings)
```

But a 0.2 s delay is already a lot better, and keeps things simpler, so I actually prefer it.

Reply:
For the record, the driver [responds](https://github.com/aleksamagicka/aquacomputer_d5next-hwmon/blob/0f525808c4267e4e1f3a28400a46171dac6f87c4/aquacomputer_d5next.c#L424) with `-ENODATA`.

Reply:
Thanks!